crates/server/src/verifysnsignature.rs

```diff
@@ -0,0 +1,313 @@
+use std::str::FromStr;
+use std::vec;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use starknet_core::types::{BlockId, BlockTag, Felt, FunctionCall};
+use starknet_core::utils::get_selector_from_name;
+use starknet_crypto::{poseidon_hash_many, verify};
+use starknet_providers::jsonrpc::HttpTransport;
+use starknet_providers::{JsonRpcClient, Provider, Url};
+
+// Structure representing a field in a StarkNet type with a name and type
+#[derive(Serialize, Deserialize, Debug)]
+struct FieldType {
+    name: String,
+    #[serde(rename = "type")]
+    field_type: String,
+}
+
+// Structure holding definitions of StarkNet types: StarknetDomain and Challenge
+#[derive(Serialize, Deserialize, Debug)]
+struct Types {
+    #[serde(rename = "StarknetDomain")]
+    stark_net_domain: Vec<FieldType>,
+    #[serde(rename = "Challenge")]
+    challenge: Vec<FieldType>,
+}
+
+// Asynchronous function to verify an Argent wallet signature on chain
+pub async fn verify_argent_signature(
+    message_hash: String,
+    signature: Vec<String>,
+    wallet_address: String,
+    message: &str,
+    rpc_node_url: &str,
+    chain_id: &str,
+) -> eyre::Result<bool> {
+    // Convert inputs from strings to StarkNet-compatible types
+    let wallet_address = Felt::from_str(&wallet_address).unwrap();
+    let message_hash = Felt::from_str(&message_hash).unwrap();
```

Comment:
let's avoid unwrapping

```suggestion
    let wallet_address = Felt::from_str(&wallet_address)?;
    let message_hash = Felt::from_str(&message_hash)?;
```

and others